### DISCUSSION TITLE
Install Package from Git Repository without adding credentials in package.json

### DISCUSSION THREAD

**Discussion by with-shrey** [npm/feedback#565] [CLI] [answered]
I was hoping if there could be a way to install from git private repository using https 
without adding username/password or token to the URL.

PHP composer uses a JSON file to define passwords 
Here is an example 
https://getcomposer.org/doc/articles/authentication-for-private-packages.md#http-basic

It would be great to have something similar in npm

**Comment by MylesBorins** [comment, marked as answer]
hey @with-shrey 

if you would like to suggest a new feature in the npm CLI the appropriate next step will be to open an issue over at the npm CLI RFC repo

https://github.com/npm/rfcs

Also worth mentioning that it is possible to configure git to use an external credential helper to avoid having to encode this information in the URL or having to enter it manually in the CLI

https://git-scm.com/docs/gitcredentials
